### PR TITLE
Simplify `run`

### LINKED ISSUE #198: `run` does not return stdout or stderr
### What happened?

See https://github.com/pulumi/pulumi/discussions/13013#discussioncomment-5996365 for initial user report on this.
I've tried repro'ing myself and can confirm that stdout and stderr are returned empty even for simple commands such as `echo hi`.

### Expected Behavior

stdout and stderr to contain the output of the command ran.

### Steps to reproduce

```
import * as command from "@pulumi/command";

// Render the Shell Script
const renderShell1 = command.local.run({
    command: "echo hello",
    environment: {
        //IMAGE_URL: imageurl,
        IMAGE_URL: "test123",
    },
  });
export const stdo = renderShell1.then(v => v.stdout);
export const stde = renderShell1.then(v => v.stderr);
```

### Output of `pulumi about`

`@pulumi/command  0.7.2`

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by iwahbe**
By moving `run` from a method on `*CommandOutputs` to it's own function, we make it obvious what it effects. This has the advantage of making it operate seamlessly on both `CommandOutputs` and `RunOutputs`.

Fixes #198

### CODE REVIEW

File: provider/pkg/provider/local/commandController.go [hunk 1]
```diff
@@ -81,6 +80,5 @@ func (c *Command) Delete(ctx p.Context, id string, props CommandOutputs) error {
 	if props.Delete == nil {
 		return nil
 	}
-	_, _, err := props.run(ctx, *props.Delete)
-	return err
+	return run(ctx, *props.Delete, &props.BaseInputs, &props.BaseOutputs)
```
Comment (mjeffryes) [inline provider/pkg/provider/local/commandController.go:83]: this is the bit that specifically fixes the linked issue right?

**Review by mjeffryes — APPROVED** [review]
I don't see any problems, and the additional test is nice, but I have to confess that I'm not entirely clear on the goal of the refactor? It looks to be pretty equivalent in complexity before and after. I guess you do get to enforce the invariant that there's a non-empty destination for stderr and stdout in a single place, but on average, functions that mutate their inputs tend to be more complicated to reason about not less. 

I'm ok with shipping this, but I think my preference would be to fix the bug without the refactor and enforce the desired behavior (of emitting stdout/stderr) at a higher level.

File: provider/pkg/provider/local/commandController.go (hunk 1 above)
Comment (iwahbe) [reply provider/pkg/provider/local/commandController.go:83]: No, the fix is provider/pkg/provider/local/runController.go:L31.

**Comment by iwahbe** [comment]
> I don't see any problems, and the additional test is nice, but I have to confess that I'm not entirely clear on the goal of the refactor? It looks to be pretty equivalent in complexity before and after. I guess you do get to enforce the invariant that there's a non-empty destination for stderr and stdout in a single place, but on average, functions that mutate their inputs tend to be more complicated to reason about not less.
> 
> I'm ok with shipping this, but I think my preference would be to fix the bug without the refactor and enforce the desired behavior (of emitting stdout/stderr) at a higher level.

The refactor/fix is necessary because `run` was a method on `*CommandOutputs`, but the `Run` invoke does't use `*CommandOutputs`. Previously we constructed a partial version of `*CommandOutputs` so we could call `run` on it, then extract the output (which we did incorrectly).

**Pull request merged**